### PR TITLE
INS-947: implement sending of PendingFinished message

### PR DESCRIPTION
**Pull request by afiskon**

### CODE REVIEW

File: logicrunner/logicrunner.go [hunk 1]
```diff
@@ -413,7 +417,38 @@ func (lr *LogicRunner) ProcessExecutionQueue(ctx context.Context, es *ExecutionS
...
+func (lr *LogicRunner) pulseChanged(ctx context.Context, es *ExecutionState, currentRef core.RecordRef) bool {
+	es.Lock()
+	defer es.Unlock()
+
+	if !es.pending {
+		return false
+	}
+
+	es.pending = false
+	msg := message.PendingFinished{Reference: currentRef}
```
Comment (ruz) [inline logicrunner/logicrunner.go]: I don't understand how this classifies as "pulseChanged".

Comment (afiskon) [reply logicrunner/logicrunner.go]: OK, fixed

**Comment by afiskon** [comment]
@ruz everything fixed according to your notes

**Comment by ruz** [comment]
Please, change the way you write unit tests. Don't change other tests. One method/function - one or more tests, not many functions in one tests.

**Comment by afiskon** [comment]
@ruz 
> Don't change other tests
Unfortunately in this particular case it would create a test that is almost an exact copy of an existing one. Which by the way is 3 tests in one.

File: logicrunner/logicrunner.go [hunk 2]
```diff
@@ -475,7 +478,40 @@ func (lr *LogicRunner) ProcessExecutionQueue(ctx context.Context, es *ExecutionS
 		}
 
 		finish()
+
+		if lr.finishPendingIfNeeded(ctx, es, *qe.parcel.Message().DefaultTarget()) {
+			// return right now just to avoid calling es.Lock() once again and figure
+			// out that the queue is empty in the beginning of the loop
+			return
+		}
+	} // for
```
Comment (aalekseev-ru) [inline logicrunner/logicrunner.go]: what this comment about?

File: logicrunner/logicrunner_unit_test.go [hunk 3]
```diff
@@ -78,6 +79,55 @@ func TestOnPulse(t *testing.T) {
...
+
+	pendingFinishedWasSent := false
+
+	mb := testutils.NewMessageBusMock(t)
+	mb.SendMock.Set(func(ctx context.Context, msg core.Message, pulse core.Pulse, opts *core.MessageSendOptions) (core.Reply, error) {
+		if msg.Type() == core.TypePendingFinished {
+			pendingFinishedWasSent = true
+		}
+		return &reply.ID{}, nil
+	})
```
Comment (ruz) [inline logicrunner/logicrunner_unit_test.go]: This can be done with Expect.

**Pull request merged**